src/Avalonia.X11/X11Platform.cs

```diff
@@ -407,13 +405,15 @@ public static AppBuilder UseX11(this AppBuilder builder)
             builder
                 .UseStandardRuntimePlatformSubsystem()
                 .UseWindowingSubsystem(() =>
-                new AvaloniaX11Platform().Initialize(AvaloniaLocator.Current.GetService<X11PlatformOptions>() ??
-                                                     new X11PlatformOptions()));
+                {
+                    var options = AvaloniaLocator.Current.GetService<X11PlatformOptions>() ?? new X11PlatformOptions();
+                    _ = new AvaloniaX11Platform(options);
```

Comment:
Maybe add a static Initialize method? Relying on ctor side effects outside of the class itself seems a bit out of place

Reply:
I've reverted to the old implementation and added `null!` initializers instead for now. We completely control the caller so it's fine.

The `AvaloniaX11Platform` instance registers itself inside the `AvaloniaLocator` currently. Ideally we should refactor that code to completely decouple initialization and registration, but that's out of scope of this PR.